### PR TITLE
Exercises issues

### PR DESCRIPTION
**Pull request by wilkus27**
All issues related with exercises, mainly from EOC.

1. #1870
2. #1956
3. #2034
4. #2026
5. #1675
6. #1983

### CODE REVIEW

File: recipes/mixins/_modify.scss [hunk 1]
```diff
@@ -927,6 +927,30 @@
...
+      .first-element {
+        string-set: hasFirstElement " has-first-element";
+      }
+      &:deferred {
+        class: attr(class) string(hasFirstElement);
+      }
+    }
+  }
+}
+
```
Comment (wilkus27) [inline recipes/mixins/_modify.scss]: There is no way to find one solution for exercises started by images/figures and those with math inside.
Either paragraph with math displays in this way:

image

or images/figures are not inline with number:

image

I had to tag exercises started with images/figures and apply for them different settings.

File: styles/designs/carnival/parts/eoc/_eoc-components2.scss [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+$Eoc__TableData: (
+    _name: "TableData",
+    _subselector: " td:not(:only-of-type)",
+    _properties:(
+        text-align: center,
+        padding-top: table-v-grid(1),
+        padding-bottom: table-v-grid(1),
+        padding-right: h-grid(.5),
+        padding-left: h-grid(.5),
```
Comment (wilkus27) [inline styles/designs/carnival/parts/eoc/_eoc-components2.scss]: I had to change paddings in tables from two columns eoc modules. I set previous values: `4px` for right and left padding. When those paddings are `0.75rem` as Alan set for all tables, some of them are too big in two columns layout:

image

That's how it looks with decreased paddings:

image

File: styles/designs/carnival/parts/eoc/_eoc-oneColumn-shapes.scss [hunk 3]
```diff
@@ -0,0 +1,24 @@
+@import '_eoc-components2';
```
Comment (wilkus27) [inline styles/designs/carnival/parts/eoc/_eoc-oneColumn-shapes.scss]: This file will be renamed in the future, when we remove old eoc files.

File: styles/templates/template1/parts/_exercises-settings.scss [hunk 4]
```diff
@@ -1,21 +1,12 @@
...
-        ),
-    ),
-    CheckUnderstandingExercises:(
-        _selectors: ('.check-understanding [data-type="exercise"]'),
-        ProblemDivider: (
-            margin-right: h-grid(1)
+    RegularExercise: (
+        _selectors: ('[data-type="exercise"]'),
+        ProblemContainerSpecific: (
+            vertical-align: bottom,
```
Comment (wilkus27) [inline styles/templates/template1/parts/_exercises-settings.scss]: This is needed to resolve https://github.com/openstax/cnx-recipes/issues/1983.

File: styles/templates/template1/parts/_exercises-settings.scss [hunk 5]
```diff
@@ -1,21 +1,12 @@
...
+            padding-top: half-v-grid(1),
```
Comment (wilkus27) [inline styles/templates/template1/parts/_exercises-settings.scss]: This is needed to resolve #1983.

File: styles/books/ap-biology/book.scss [hunk 6]
```diff
@@ -57,8 +63,8 @@
 @include use('TeacherMaterial', 'common:::Hidden');
 @include use('APEverydayConnectionNote','carnival:::BoxedNoteShape');
 @include use('ColumnHeaderTable', 'carnival:::ColumnHeaderTableShape');
-@include use('Review', 'carnival:::EocExerciseWithoutSubtitle');
-@include use('TestPrep', 'carnival:::EocExerciseWithoutSubtitle');
-@include use('SciencePractice', 'carnival:::EocExerciseWithoutSubtitle');
+@include use('Review', 'carnival:::EocTwoColumnsUnclustered');
```
Comment (helenemccarron) [inline styles/books/ap-biology/book.scss]: I like this renaming convention, I can tell right away what it does 👍

**Pull request merged**